extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java

```diff
@@ -503,6 +510,33 @@ public Key resolveKey(JsonWebSignature jws, List<JsonWebStructure> nestingContex
             }
         }
 
+        private Key findKeyByBruteForce(JsonWebSignature jws) {
+            final Key initialKey = jws.getKey();
+            final String keyType;
+
+            try {
+                keyType = jws.getKeyType();
+            } catch (InvalidAlgorithmException e) {
+                return null;
+            }
+
+            for (Key key : jwks.getKeysForKeyType(keyType)) {
+                jws.setKey(key);
+
+                try {
+                    if (jws.verifySignature()) {
+                        jws.setKey(initialKey);
+                        return key;
+                    }
+                } catch (JoseException e) {
+                    LOG.debug("Verifying signature with key " + key + " failed.", e);
```

Comment:
Please use `debugf`. 
It would also likely make it easier to understand that these are insignificant failures if you add a log message above saying that a signature verification check is used to find a matching key and confirm the result (key found or not)

Reply:
Added more elaborate logging into the method.